src/App.jsx

```diff
@@ -13,34 +14,96 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const defaultState = {
+  reversed: false,
+  isLength: false,
+  alphabet: false,
+};
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [trackStatus, setTrackStatus] = useState(defaultState);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const { reversed, isLength, alphabet } = trackStatus;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const getGoods = () => {
+    const goods = [...goodsFromServer];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (isLength || alphabet) {
+      goods.sort((a, b) => {
+        if (isLength) {
+          return a.length - b.length;
+        }
+
+        return a.localeCompare(b);
+      });
+    }
+
+    if (reversed) {
+      goods.reverse();
+    }
+
+    return goods;
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${alphabet ? '' : 'is-light'}`}
+          onClick={() =>
+            setTrackStatus({
+              ...trackStatus,
+              alphabet: !alphabet,
+              isLength: false,
+            })
+          }
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={`button is-success ${isLength ? '' : 'is-light'}`}
```

Comment:
Using the classNames library will improve readability and help support potential future changes.

Example:

```suggestion
className={classNames('button', 'is-success', {
  'is-light': !isLength,
})}
```